keras/activations.py

```diff
@@ -4,14 +4,24 @@
 from .utils.generic_utils import deserialize_keras_object
 
 
-def softmax(x):
+def softmax(x, axis=-1):
+    """
+
+    # Arguments
+        x : tensor
+        axis: integer, defines along which axis the normalization is applied
+    """
     ndim = K.ndim(x)
     if ndim == 2:
         return K.softmax(x)
     elif ndim == 3:
         e = K.exp(x - K.max(x, axis=-1, keepdims=True))
         s = K.sum(e, axis=-1, keepdims=True)
         return e / s
+    elif ndim == 4:
```

Comment:
You don't need this case. It's just either 2D -> `K.softmax`, > 2D -> use `axis`.